### PR TITLE
feat(capnp): improve queries

### PR DESCRIPTION
**Pull request by amaanq**
I rewrote the grammar a bit to distinguish escape sequences and fix a couple bugs; those changes are reflected in the highlights

I also added indents, injections, and locals

### CODE REVIEW

File: queries/capnp/highlights.scm [hunk 1]
```diff
@@ -68,8 +58,7 @@
 
 (enum_identifier) @type
 (extend_type) @type
-(field_type) @type
-(generic_identifier) @type
+; (generic_identifier) @type
```
Comment (ObserverOfTime) [inline queries/capnp/highlights.scm:61]: Why is this commented out?

Comment (amaanq) [reply queries/capnp/highlights.scm:61]: whoops, forgot to remove. generics *can* contain other types nested inside them, so it's redundant and would capture parentheses as `@type` sometimes.

File: queries/capnp/highlights.scm [hunk 2]
```diff
@@ -95,11 +94,16 @@
...
+  (string)
+  (concatenated_string)
+  (block_text)
+  (namespace)
+] @string
+
+(escape_sequence) @string.escape
 
 (data_string) @string.special
-(namespace) @string.special
```
[the remark this answers is not in the document]
Comment (amaanq) [reply queries/capnp/highlights.scm:102]: `$Cxx.namespace("capnproto_test::capnp::test");`

where the string inside the parentheses was the namespace..somewhat a mistake I made to think it was a special string, well, a c++ injection here would be cool but somewhat useless

**Comment by amaanq** [comment]
wondering if

```scm
(annotation
  (generics
    (generic_parameters
      (generic_identifier) @definition.type)))

(replace_using
  (generics
	(generic_parameters
	  (generic_identifier) @definition.type)))

(return_type
  (generics
	(generic_parameters
	  (generic_identifier) @definition.type)))

(custom_type
  (generics
	(generic_parameters
	  (generic_identifier) @definition.type)))
```

can be flattened...would be nice

**Comment by ObserverOfTime** [comment]
Can't you just remove the top node?

```clojure
  (generics
	(generic_parameters
	  (generic_identifier) @definition.type))
```

**Comment by amaanq** [comment]
> Can't you just remove the top node?
> 
> ```clojure
>   (generics
> 	(generic_parameters
> 	  (generic_identifier) @definition.type))
> ```

No..sometimes it's a parameter and sometimes a reference depending on where it's used

```scm
(custom_type
  (generics
    (generic_parameters 
      (generic_identifier) @reference)))

(struct
  (generics
    (generic_parameters
      (generic_identifier) @definition.parameter)))

(interface
  (generics
    (generic_parameters
      (generic_identifier) @definition.parameter)))

(method
  (implicit_generics
    (implicit_generic_parameters
      (generic_identifier) @definition.parameter)))

(method
  (generics
    (generic_parameters
      (generic_identifier) @definition.parameter)))
```

**Comment by ObserverOfTime** [comment]
This may or may not work but I'm not sure if it's even a better solution:

```clojure
([(annotation) (replace_using) (return_type) (custom_type)]
  (generics
    (generic_parameters
      (generic_identifier) @definition.type)))
```

Or, you can cheat a little and define a visible [supertype](https://tree-sitter.github.io/tree-sitter/using-parsers#supertype-nodes) for them in the grammar.

**Comment by amaanq** [comment]
Adding a supertype doesn't fit imo, these are really unrelated rules and would feel clunky at best.

I think it's fine as is, not sure if your suggestion is better (I thought of that too)

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**